components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java

```diff
@@ -833,6 +853,31 @@ public static String getUserConsentURL(OAuth2Parameters params, String loggedInU
         return consentPageUrl;
     }
 
+    private static String getScopeMetadataQueryParam(Set<String> scopes, String tenantDomain) {
+
+        try {
+            List<String> oidcScopeList = oAuthAdminService.getRegisteredOIDCScope(tenantDomain);
+            List<String> nonOidcScopeList = new ArrayList<>();
+
+            for (String scope : scopes) {
+                if (!oidcScopeList.contains(scope)) {
+                    nonOidcScopeList.add(scope.toLowerCase());
```

Comment:
Scope are case sensitive according to [OAuth Specification](https://www.rfc-editor.org/rfc/rfc6749#section-3.3). Shall we remove the toLowerCase().